src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -577,20 +578,22 @@ public void onNewScrollContext(ReaderContext readerContext) {
 
                 @Override
                 public void validateReaderContext(ReaderContext readerContext, TransportRequest transportRequest) {
-                    final Object _isLocal = readerContext.getFromContext("_opendistro_security_scroll_auth_local");
-                    final Object _user = readerContext.getFromContext("_opendistro_security_scroll_auth");
-                    if (_user != null && (_user instanceof User)) {
-                        final User scrollUser = (User) _user;
-                        final User currentUser = threadPool.getThreadContext()
-                                .getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
-                        if (!scrollUser.equals(currentUser)) {
+                    if (transportRequest instanceof InternalScrollSearchRequest) {
+                        final Object _isLocal = readerContext.getFromContext("_opendistro_security_scroll_auth_local");
+                        final Object _user = readerContext.getFromContext("_opendistro_security_scroll_auth");
+                        if (_user != null && (_user instanceof User)) {
+                            final User scrollUser = (User) _user;
+                            final User currentUser = threadPool.getThreadContext()
+                                    .getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+                            if (!scrollUser.equals(currentUser)) {
+                                auditLog.logMissingPrivileges(SearchScrollAction.NAME, transportRequest, null);
+                                log.error("Wrong user {} in reader context, expected {}", scrollUser, currentUser);
+                                throw new ElasticsearchSecurityException("Wrong user in reader context", RestStatus.FORBIDDEN);
```

Comment:
should it be scroll context or reader context?

Reply:
We expect `ReaderContext` (former `SearchContext`) to be populated. The check is done only for scroll (`InternalScrollSearchRequest`).